src/recon_buildblock/CUDA_stir/CudaRelativeDifferencePrior.cu

```diff
@@ -0,0 +1,382 @@
+#include "stir/recon_buildblock/CUDA_stir/CudaRelativeDifferencePrior.h"
+#include "stir/Succeeded.h"
+#include "stir/recon_buildblock/GeneralisedPrior.h"
+#include "stir/DiscretisedDensityOnCartesianGrid.h"
+#include "stir/IndexRange3D.h"
+#include "stir/is_null_ptr.h"
+
+extern "C" __global__
+void computeCudaRelativeDifferencePriorGradientKernel(float* tmp_grad,
+                                                    const float* image,
+                                                    const float* weights,
+                                                    const float* kappa,
+                                                    const bool do_kappa,
+                                                    const float gamma,
+                                                    const float epsilon,
+                                                    const int z_dim,
+                                                    const int y_dim,
+                                                    const int x_dim) {
+    // tmp_grad, cp_image, cp_weights, cp_kappa, cp_penalisation_factor, cp_gamma, cp_epsilon, z_dim, y_dim, x_dim
+    int z = blockIdx.z * blockDim.z + threadIdx.z;
+    int y = blockIdx.y * blockDim.y + threadIdx.y;
+    int x = blockIdx.x * blockDim.x + threadIdx.x;
```

Comment:
add `const` here (I don't think they're changing), and a comment what they are

Reply:
done here: https://github.com/UCL/STIR/pull/1408/commits/f41728e6c06347fa84889cb355dcfc9a5c4d5af8